lib/ansible/modules/system/hostname.py

```diff
@@ -422,40 +398,33 @@ class OpenRCStrategy(GenericStrategy):
     def get_permanent_hostname(self):
         name = 'UNKNOWN'
         try:
-            try:
-                f = open(self.HOSTNAME_FILE, 'r')
+            with open(self.HOSTNAME_FILE, 'r') as f:
                 for line in f:
                     line = line.strip()
                     if line.startswith('hostname='):
                         name = line[10:].strip('"')
                         break
-            except Exception as e:
-                self.module.fail_json(msg="failed to read hostname: %s" %
-                                          to_native(e), exception=traceback.format_exc())
-        finally:
-            f.close()
+        except Exception as e:
+            self.module.fail_json(msg="failed to read hostname: %s" %
+                                      to_native(e), exception=traceback.format_exc())
```

Comment:
This changes behavior slightly (exception is now caught when error happens during closing), but I think that's ok.